user-guide/Advanced_Functionality/Databases/Configuring_dedicated_clustered_storage/Indexing_database/OpenSearch_Database/Installing_OpenSearch_database.md

```diff
@@ -21,13 +24,286 @@ See the [official documentation](https://opensearch.org/docs/latest/) on how to
+  #
+  cluster.name: NameOfYourCluster
+  cluster.initial_master_nodes: ["opensearchnode1"]
+  #
+  # ------------------------------------ Node ------------------------------------
+  #
+  # Use a descriptive name for the node(node.name can be anything you desire, but we recommend hostname of node):
+  #
+  # Make the node.name reflect the name of your node
+  node.name: opensearchnode1
+  #
+  # ----------------------------------- Paths ------------------------------------
+  #
+  # Path to directory where to store the data (separate multiple locations by comma):
+  #
+  path.data: /var/lib/opensearch
+  #
+  # Path to log files:
+  #
+  path.logs: /var/log/opensearch
+  # ---------------------------------- Network -----------------------------------
+  #
+  # Set the bind address to a specific IP (IPv4 or IPv6), best is to use the real-ip of the node:
+  #
+  network.host: ipAddressOfTheNode
+  
+  network.publish_host: ipAddressOfTheNode
+  #
+  # Set a custom port for HTTP:
+  #
+  http.port: 9200
+  #
+  # For more information, consult the network module documentation.
+  #
+  # --------------------------------- Discovery ----------------------------------
+  #
+  # Pass an initial list of hosts to perform discovery when this node is started:
+  # The default list of hosts is ["127.0.0.1", "[::1]"]
+   #
+  discovery.seed_hosts: ["ipAddressOfTheFirstNode","ipAddressOfTheSecondNode","ipAddressOfTheThirdNode"]
+  
+  discovery.type: zen
+  ```
+
+- If you want a node to be a **data node**, add the following configuration in *OpenSearch.yml*:
+
+  ```yml
+  node.roles: [ data, ingest ]
+  ```
+
+- If you want a node to be the **cluster manager node** (a.k.a. the master node), add the following configuration in *OpenSearch.yml*:
+
+  ```yml
+  node.roles: [ cluster_manager ]
+  ```
+
+#### TLS configuration
+
+To configure TLS, instead of using .pem files, we recommend generating p12 keystore and truststore .p12 files using the [Generate-TLS-Certificates](https://github.com/SkylineCommunications/generate-tls-certificates) script on GitHub. This script will generate a .p12 file for every node. Place the correct .p12 file on each corresponding node. The GitHub script will provide you with the keystore and truststore password. Add this to the keystore and truststore password fields in the OpenSearch.yml file. A rootCA.crt will also be generated by the GitHub script.
+
+> [!NOTE]
+> In case you want to query the OpenSearch database from a different computer, you can also install the rootCA.crt on a Windows machine by double-clicking it. Use *Local Machine* as *Store Location* when adding the certificate, and place the certificate in *Trusted Root Certification Authorities*. This is similar to the [Client-server TLS encryption](xref:Security_Elasticsearch#client-server-tls-encryption) procedure for Elasticsearch. Once the certificate has been imported, there is trust between your machine running DataMiner and the OpenSearch database/cluster.
```

Comment:
Why would you need to do this? Is it optional or is it a must? Could you explain this? Also, if it is a must, could you add it as part of the procedure instead of in a separate note?

Reply:
> Why would you need to do this? Is it optional or is it a must? Could you explain this? Also, if it is a must, could you add it as part of the procedure instead of in a separate note?

It is a must for DataMiner to be able to poll/connect to OpenSearch